pystac/stac_object.py

```diff
@@ -521,6 +522,7 @@ def from_file(cls, href):
         """
         if not is_absolute_href(href):
             href = make_absolute_href(href)
+        href = f"file:{pathname2url(href)}"
```

Comment:
This breaks tests - perhaps a better approach would be to put this line directly in the STAC_IO method you point to in the body of the PR?

Reply:
Let me try to work out a different way of doing this. In my case, all test pass with this fix, so I guess the behaviour in the CI and in my machine (or any windows machine, for that matter), might be different.